clients/src/test/java/org/apache/kafka/common/record/FileLogInputStreamTest.java

```diff
@@ -209,6 +209,14 @@ public void testBatchIterationIncompleteBatch() throws IOException {
         }
     }
 
+    @Test
+    public void testNextBatchSelectionAtMaxIntValue() throws IOException {
+        if (magic == 0 && compression == CompressionType.NONE) {
```

Comment:
Seems no harm running this test case for every parameterization. I know it's redundant since we are not using either of the fields, but it feels less arbitrary.